From 96056056ad0f331ddf40b732b5288fa7f4ab5754 Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Sun, 7 Mar 2021 17:30:58 -0500
Subject: [PATCH] New net magic for main, test and regtest (#50)

Rebase provenance: this historical change was adapted onto the pinned
Bitcoin ABC base. Source diff is the recorded Lotus V1 replay commit below.

Historical-source-commit: 220efe8dd36bcb5a8efb09ae103aa976bc025e06
Recorded-replay-commit: f7648d289b721dee602427921d10c3e5aef9b060

diff --git a/src/kernel/chainparams.cpp b/src/kernel/chainparams.cpp
index bfa3bf3353..4ac92c509b 100644
--- a/src/kernel/chainparams.cpp
+++ b/src/kernel/chainparams.cpp
@@ -161,10 +161,10 @@ public:
         diskMagic[1] = 0xbe;
         diskMagic[2] = 0xb4;
         diskMagic[3] = 0xd9;
-        netMagic[0] = 0xe3;
-        netMagic[1] = 0xe1;
-        netMagic[2] = 0xf3;
-        netMagic[3] = 0xe8;
+        netMagic[0] = 0xec;
+        netMagic[1] = 0xe7;
+        netMagic[2] = 0xef;
+        netMagic[3] = 0xf3;
         nDefaultPort = 8333;
         nPruneAfterHeight = 100000;
         m_assumed_blockchain_size =
@@ -327,8 +327,8 @@ public:
         diskMagic[1] = 0x11;
         diskMagic[2] = 0x09;
         diskMagic[3] = 0x07;
-        netMagic[0] = 0xf4;
-        netMagic[1] = 0xe5;
+        netMagic[0] = 0xec;
+        netMagic[1] = 0xf4;
         netMagic[2] = 0xf3;
         netMagic[3] = 0xf4;
         nDefaultPort = 18333;
@@ -480,10 +480,10 @@ public:
         diskMagic[1] = 0xbf;
         diskMagic[2] = 0xb5;
         diskMagic[3] = 0xda;
-        netMagic[0] = 0xda;
-        netMagic[1] = 0xb5;
-        netMagic[2] = 0xbf;
-        netMagic[3] = 0xfa;
+        netMagic[0] = 0xec;
+        netMagic[1] = 0xf2;
+        netMagic[2] = 0xe5;
+        netMagic[3] = 0xe7;
         nDefaultPort = 18444;
         nPruneAfterHeight = opts.fastprune ? 100 : 1000;
         m_assumed_blockchain_size = 0;
diff --git a/test/functional/test_framework/p2p.py b/test/functional/test_framework/p2p.py
index 60eb0b5577..b331ad6543 100644
--- a/test/functional/test_framework/p2p.py
+++ b/test/functional/test_framework/p2p.py
@@ -147,9 +147,9 @@ MESSAGEMAP = {
 }
 
 NET_MAGIC_BYTES = {
-    "mainnet": b"\xe3\xe1\xf3\xe8",
-    "testnet3": b"\xf4\xe5\xf3\xf4",
-    "regtest": b"\xda\xb5\xbf\xfa",
+    "mainnet": b"\xec\xe7\xef\xf3",
+    "testnet3": b"\xec\xf4\xf3\xf4",
+    "regtest": b"\xec\xf2\xe5\xe7",
 }
 
 DISK_MAGIC_BYTES = {
